CMake/ClangTidy.cmake

```diff
@@ -0,0 +1,31 @@
+if (clang_tidy)
+
+  function (find_clang_tidy _CLANG_TIDY_PATH)
+    if (DEFINED ENV{CLIO_CLANG_TIDY_BIN})
+      set (_CLANG_TIDY_PATH $ENV{CLIO_CLANG_TIDY_BIN})
+      if (NOT EXISTS ${_CLANG_TIDY_PATH})
+        message (FATAL_ERROR "$ENV{CLIO_CLANG_TIDY_BIN} no such file. Check CLIO_CLANG_TIDY_BIN env variable")
+      endif ()
+      message (STATUS "Using clang-tidy from CLIO_CLANG_TIDY_BIN")
+    else ()
+      find_program (_CLANG_TIDY_PATH NAMES "clang-tidy-16" "clang-tidy-15" "clang-tidy-14" "clang-tidy" REQUIRED)
+    endif ()
+    if (NOT _CLANG_TIDY_PATH)
+      message (FATAL_ERROR
+        "clang-tidy binary not found. Please set the CLIO_CLANG_TIDY_BIN environment variable or install clang-tidy.")
+    endif ()
+  endfunction ()
+
+  # Support for https://github.com/matus-chochlik/ctcache
+  find_program (CLANG_TIDY_CACHE_PATH NAMES "clang-tidy-cache")
+  if (CLANG_TIDY_CACHE_PATH)
+    find_clang_tidy (_CLANG_TIDY_PATH)
```

Comment:
find_clang_tidy can run outside the if statement? Do we need this variable :CLANG_TIDY_PATH ?